demo/verifier.ts

```diff
@@ -0,0 +1,66 @@
+import { Hasher, KeyBindingVerifier, base64encode, decodeJWT } from '@meeco/sd-jwt';
+import { createHash } from 'crypto';
+import { JWK, KeyLike, importJWK, jwtVerify } from 'jose';
+import { Verifier, defaultHashAlgorithm, supportedAlgorithm } from '../dev/src';
+
+function verifierCallbackFn(publicKey: Uint8Array | KeyLike) {
+  return async (jwt: string): Promise<boolean> => {
+    const verifiedKbJWT = await jwtVerify(jwt, publicKey);
+    return !!verifiedKbJWT;
+  };
+}
+
+function hasherCallbackFn(alg: string = defaultHashAlgorithm): Hasher {
+  return (data: string): string => {
+    const digest = createHash(alg).update(data).digest();
+    return base64encode(digest);
+  };
+}
+
+function kbVeriferCallbackFn(expectedAud: string, expectedNonce: string): KeyBindingVerifier {
+  return async (kbjwt: string, holderJWK: JWK) => {
+    const { header, payload } = decodeJWT(kbjwt);
+
+    if (expectedAud || expectedNonce) {
+      if (payload.aud !== expectedAud) {
+        throw new Error('aud mismatch');
+      }
+      if (payload.nonce !== expectedNonce) {
+        throw new Error('nonce mismatch');
+      }
+    }
+
+    if (!Object.values(supportedAlgorithm).includes(header.alg as supportedAlgorithm)) {
+      throw new Error('unsupported algorithm');
+    }
+
+    const holderKey = await importJWK(holderJWK, header.alg);
+    const verifiedKbJWT = await jwtVerify(kbjwt, holderKey);
+    return !!verifiedKbJWT;
+  };
+}
+
+async function main() {
+  const verifier = new Verifier();
+  const { vcSDJWTWithkeyBindingJWT, nonce } = {
```

Comment:
IS there a reason to have such notation instead of two separate:
```
const vcSDJWTWithkeyBindingJWT = '...';
const nonce = '...';
```
?

Reply:
it was due to my ignorance, will fix it. thanks